### DISCUSSION TITLE
Is there an automatic navigation for pages within folder?

### DISCUSSION THREAD

**Discussion by buzzer70** [datenstrom/community#277] [Ask a question] [answered]
Hi, great little CMS by the way.
Is there a plugin / command that will allow me to display a navigation bar for the files in the current folder? I add a page using the + button and can obviously create a link to the page but wondering if i was missing out on a way to do this already so if i add or remove a page, the nav automatically updates - **Thanks**

**Comment by nibreh** [comment, marked as answer]
HI

I'm not sure if I understand, but maybe this can help you:

1. In the settings of your page (or in `config.ini` for all site) you can use

`Navigation: navigation-tree`

2. Take a look at the [plugin preview](https://github.com/datenstrom/yellow-plugins/tree/master/preview). You can list all pages from a folder with shorcut 

`[preview /folder/ simple]`

for example in a sidebar.

Cheers

**Comment by buzzer70** [comment]
Hi 

Thanks for the quick response, you understood pretty well, i had not looked at the navigation-tree so thanks for that, it will probably be fine for what i need.

What i had imagined was to list the tree part like the navigation.

**Comment by buzzer70** [comment]
Hey - i created a plug in to do what i wanted after your help. It creates a list of page links in a list which is styled and uses a fontawesome icon, is this of interest to the project? I'm new to git so dont know the procedure for pulling / pushing etc. but am happy to share what i have done for others to use - i have included a screenshot using the "fa-folder-o" fontawesome icon. 
This is the markdown:

[pagelinks /photos/ fa-folder-o ]

![capture](https://user-images.githubusercontent.com/30840821/29194817-4127a0c6-7e23-11e7-99ac-58f1aabf9ac5.PNG)

**Comment by markseuffert** [comment]
Hey, just a few days with Datenstrom Yellow and already wrote your own plugin. Here are some tips:

* https://datenstrom.se/yellow/help/contributing-guidelines
* https://github.com/annaesvensson/yellow-publish

Contributions and feedback are very welcome.
